contrib/inventory/stacki.py

```diff
@@ -0,0 +1,187 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+# Stacki inventory script
+# Configure stacki.yml with proper auth information and place in the following:
+#  - ../inventory/stacki.yml
+#  - /etc/stacki/stacki.yaml
+#  - /etc/ansible/stacki.yml
+# The stacki.yaml file can contain entries for authentication information
+# regarding the Stacki front-end node.
+#
+# use_hostnames uses hostname rather than interface ip to as connection
+#
+#
+"""
+
+Example Usage:
+    List Stacki Nodes
+    $ ./stack.py --list
+
+
+Example Configuration:
+---
+stacki:
+  auth:
+    stacki_user: admin
+    stacki_password: abc12345678910
+    stacki_endpoint: http://192.168.200.50/stack
+use_hostnames: false
+
+
+"""
+
+
+
+import argparse
+import collections
+import os
+import sys
+import requests
+import yaml
+import time
+from distutils.version import StrictVersion
+
+try:
+    import json
+except:
+    import simplejson as json
+
+
+CONFIG_FILES = ['/etc/stacki/stacki.yaml', '/etc/ansible/stacki.yml']
+
+
+def stack_auth(params):
+    endpoint    = params['stacki_endpoint']
+    auth_creds  = {'USERNAME': params['stacki_user'],
+                   'PASSWORD': params['stacki_password']}
+
+    client = requests.session()
+    client.get(endpoint)
+
+    init_csrf = client.cookies['csrftoken']
+
+    header = {'csrftoken': init_csrf, 'X-CSRFToken': init_csrf,
+              'Content-type': 'application/x-www-form-urlencoded'}
+
+    login_endpoint = endpoint + "/login"
+
+    login_req = client.post(login_endpoint, data=auth_creds, headers=header)
+
+    csrftoken = login_req.cookies['csrftoken']
+    sessionid = login_req.cookies['sessionid']
+
+    auth_creds.update(CSRFTOKEN=csrftoken, SESSIONID=sessionid)
+
+    return client, auth_creds
+
+
+def stack_build_header(auth_creds):
+    header = {'csrftoken': auth_creds['CSRFTOKEN'],
+              'X-CSRFToken': auth_creds['CSRFTOKEN'],
+              'sessionid': auth_creds['SESSIONID'],
+              'Content-type': 'application/json'}
+
+    return header
+
+
+def stack_host_list(endpoint, header, client):
+
+    stack_r = client.post(endpoint, data=json.dumps({ "cmd": "list host"}),
+                          headers=header)
+    import ast
+
+    return json.loads(ast.literal_eval(stack_r.text))
+
+
+def stack_net_list(endpoint, header, client):
+
+    stack_r = client.post(endpoint, data=json.dumps({ "cmd": "list host interface"}),
+                          headers=header)
+    import ast
+
+    return json.loads(ast.literal_eval(stack_r.text))
+
+
+def format_meta(hostdata, intfdata, config):
+    use_hostnames = config['use_hostnames']
+    meta = dict(all=dict(hosts=list()),
+                frontends=dict(hosts=list()),
+                backends=dict(hosts=list()),
+                _meta=dict(hostvars=dict()))
+    for host in hostdata:
+        del host['environment']
```

Comment:
Not sure it's safe to delete keys while iterating through a dictionary.  `for host in hostdata.keys()` is safer as it creates a new list to iterate over.